contrib/kafka/eventsource/cmd/eventsource/main.go

```diff
@@ -0,0 +1,169 @@
+	config.Net.SASL.Password = eventsourceconfig.NetSaslPassword
+	config.Consumer.MaxWaitTime = time.Duration(eventsourceconfig.ConsumerMaxWaitTime)
+	config.Consumer.MaxProcessingTime = time.Duration(eventsourceconfig.ConsumerMaxProcessingTime)
+	config.Consumer.Offsets.CommitInterval = time.Duration(eventsourceconfig.ConsumerOffsetsCommitInterval)
+	config.Consumer.Offsets.Retention = time.Duration(eventsourceconfig.ConsumerOffsetsRetention)
+	config.Consumer.Offsets.Retry.Max = int(eventsourceconfig.ConsumerOffsetsRetryMax)
+	config.ChannelBufferSize = int(eventsourceconfig.ChannelBufferSize)
+	config.Group.Session.Timeout = time.Duration(eventsourceconfig.GroupSessionTimeout)
+
+	kafkaversion, err := sarama.ParseKafkaVersion(eventsourceconfig.KafkaVersion)
+	if err != nil {
+		log.Println("Unsupported Kafka Version. Should be in format 2.0.0")
+		log.Printf("Defaulting to minimum supported version: %s", kafkaversion)
+	}
+	config.Version = kafkaversion
+	log.Printf("Setting Kafka version to: %s", config.Version)
+
+	config.Consumer.Offsets.Initial = sarama.OffsetNewest
+	if eventsourceconfig.ConsumerOffsetsInitial == "OffsetOldest" {
+		log.Println("Setting to offsetOldest")
+		config.Consumer.Offsets.Initial = sarama.OffsetOldest
+	}
+
+	config.Group.PartitionStrategy = cluster.StrategyRange
+	if eventsourceconfig.GroupPartitionStrategy == "RoundRobin" {
+		config.Group.PartitionStrategy = cluster.StrategyRoundRobin
+	}
+
+	// init consumer
+	brokers := []string{eventsourceconfig.BootStrapServers}
+	topics := []string{eventsourceconfig.KafkaTopic}
+
+	consumerGroupID := eventsourceconfig.ConsumerGroupID
+	if consumerGroupID == "" {
+		consumerGroupID = uuid.New().String()
+	}
+	log.Printf("Setting ConsumerGroupID to: %s", consumerGroupID)
+
+	config.Consumer.Return.Errors = false
+
+	consumer, err := cluster.NewConsumer(brokers, consumerGroupID, topics, config)
+	if err != nil {
+		panic(err)
+	}
+	defer consumer.Close()
+
+	// trap SIGINT to trigger a shutdown.
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt)
+
+	// consume notifications
+	go func() {
+		for ntf := range consumer.Notifications() {
+			log.Printf("Rebalanced: %+v\n", ntf)
+		}
+	}()
+
+	// consume messages, watch signals
+	for {
+		select {
+		case msg, ok := <-consumer.Messages():
+			if ok {
+				// fmt.Fprintf(os.Stdout, "%s/%d/%d\t%s\t%s\n", msg.Topic, msg.Partition, msg.Offset, msg.Key, msg.Value)
+				log.Printf("Received %s", msg.Value)
+
+				var jsonPayload map[string]interface{}
+				err := json.Unmarshal(msg.Value, &jsonPayload)
+				if err != nil {
+						log.Printf("Unable to unmarshal message value: %s", msg.Value)
+						log.Printf("Error: %s", err)
+						//todo: Fix error catching as above doesn't fire
+						posterr := postMessage(msg.Key, msg.Timestamp, eventsourceconfig.KafkaTopic, eventsourceconfig.Target, msg.Partition, msg.Offset, "application/json", msg.Value)
+						if posterr != nil{
```

Comment:
should here be `posters == nil` instead?

Reply:
Good catch!